### PR TITLE
Adding asyncgenerator and asyncgeneratorfunction

### PR DESCRIPTION
**Pull request by Rumyra**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes

Adding asyncgenerator and asyncgeneratorfunction as these pages are to be added to MDN https://github.com/mdn/content/pull/2861

- [x] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)

Assuming async generator methods can not be implemented without these (see https://github.com/mdn/browser-compat-data/blob/master/javascript/functions.json) I have added these release versions

- [x] Data: if you tested something, describe how you tested with details like browser and version

I have not tested this other than async generator functions appear to work in Chrome, Safari, FF & Opera, however I only have the latest versions (if there's a better way to test this pls let me know)

- [x] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)

Tests pass

- [x] Link to related issues or pull requests, if any

See above PR

### CODE REVIEW

**Comment by ddbeck** [comment]
@Elchi3 can you provide some guidance on the structure here?

We have some of these fictitious global objects (e.g., `javascript.builtins.Generator`), but we've also rejected some in the past (e.g., #5519). There's also a distinction to be made between, for example `javascript.builtins.Generator` and `javascript.statements.generator_function`, that I don't understand.

Naïvely, I'd say that `AsyncGenerator` doesn't actually exist, even though the subfeatures are plainly of interest. Not sure how to proceed here.

**Comment by Elchi3** [comment]
> but we've also rejected some in the past (e.g., #5519).

That was about https://github.com/tc39/proposal-iterator-helpers which is still stage 2 only and as far as I can tell it is still too early to add it.

I believe this PR is about async iterators which is in ES2018 and shipped (https://github.com/tc39/proposal-async-iteration)

**Comment by Elchi3** [comment]
> There's also a distinction to be made between, for example `javascript.builtins.Generator` and `javascript.statements.generator_function`, that I don't understand.

`javascript.builtins.Generator` is https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator which is an object is returned by a generator function.

`javascript.statements.generator_function` is https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function* which is the `function*` declaration.

File: javascript/builtins/AsyncGenerator.json [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        },
+        "next": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Generator/next",
+            "spec_url": "https://tc39.es/ecma262/#sec-generator.prototype.next",
```
Comment (Elchi3) [inline javascript/builtins/AsyncGenerator.json:69]:
```suggestion
            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator/next",
            "spec_url": "https://tc39.es/ecma262/#sec-asyncgenerator-prototype-next",
```

File: javascript/builtins/AsyncGenerator.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "return": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Generator/return",
+            "spec_url": "https://tc39.es/ecma262/#sec-generator.prototype.return",
```
Comment (Elchi3) [inline javascript/builtins/AsyncGenerator.json:132]:
```suggestion
            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator/return",
            "spec_url": "https://tc39.es/ecma262/#sec-asyncgenerator-prototype-return",
```

File: javascript/builtins/AsyncGenerator.json [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "throw": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Generator/throw",
+            "spec_url": "https://tc39.es/ecma262/#sec-generator.prototype.throw",
```
Comment (Elchi3) [inline javascript/builtins/AsyncGenerator.json:195]:
```suggestion
            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator/throw",
            "spec_url": "https://tc39.es/ecma262/#sec-asyncgenerator-prototype-throw",
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
MDN and specification URLs need fixing

**Comment by Elchi3** [comment]
There is also https://tc39.es/ecma262/#sec-async-generator-function-definitions

This would be a new entry `javascript.statements.async_generator_function` and a new page https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*
It would be analog to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*

Same for `javascript.operators.async_function_star` and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/async_function*
Analog to: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function*

**Review by Elchi3 — APPROVED** [review]
Very nice work! 👍

**Comment by Rumyra** [comment]
Thanks so much for the pointers @Elchi3 ☺️

Docs are incoming 👍

**Pull request merged**